v-next/core/src/types/global-options.ts

```diff
@@ -25,7 +29,7 @@ export interface GlobalOption<T extends ParameterType = ParameterType> {
  * Runtime Environment.
  */
 // eslint-disable-next-line @typescript-eslint/no-empty-interface -- To be used through module augmentation
-export interface GlobalOptions {}
+export interface GlobalOptions extends Record<string, ParameterValue> {}
```

Comment:
I think this will be a bit annoying, as you can do `globalOptions.thisDoesntExist` and you wouldn't get a type error

Reply:
What motivated it? We may find an alternative

Reply:
If I remember correctly, this was done to avoid an error when assigning the parsed value to the global options. As the interface is empty, this throws a TS error: `globalOptions[name] = parsedValue;`

> Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'GlobalOptions'.
>   No index signature with a parameter of type 'string' was found on type 'GlobalOptions'.ts(7053)

Reply:
Fixed in https://github.com/NomicFoundation/hardhat/pull/5414/commits/22abbd50240a9fcb3ac488c72b64b5dc0fce4ec8